pkg/gen/nodejs-templates/kind.ts.mustache

```diff
@@ -52,10 +53,29 @@ import { getVersion } from "../../version";
        * @param opts A bag of options that control this resource's behavior.
        */
       constructor(name: string, args?: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.CustomResourceOptions) {
+          // In Kubernetes, `.metadata.namespace` must be a `string`. In Pulumi, we allow users to
+          // pass in `string | core.v1.Namespace`. Thus, if the user has provided a
+          // `core.v1.Namespace`, we convert it to a `string` representing that namespace's name.
+          args &&
```

Comment:
I think that this code just won't work at all:
- The result of the `apply` is never awaited
- Mutating the inputs is probably not a good idea (what if they are shared amongst multiple resources?) if it works at all (and it may not, cc @lukehoban)
- Because `core.v1.Namespace.metadata.name` is itself an output, we may lose dependency information if it is injected inside an `apply` (not 100% sure here, but this definitely happens when an output is _resolved_ to another output)

Reply:
I think this should work because the `name` is guaranteed to be present after the `Check`, but you're right that this is not acceptable to ship. Let's fix.

Reply:
I've cleaned this up a bit -- I'm still on the fence about whether we should move `unpackNamespace` out into a utility function in some other file. I'm less concerned that normal because it's all code-gen'd, but if there was a good place to put it I'd be open to moving it out.